### DISCUSSION TITLE
Couple additions to AuthenticatorInterface

### DISCUSSION THREAD

**Discussion by fch4um0n7** [codeigniter4/shield#1233] [General]
Hey all, 

I thought it'd be helpful to suggest adding the following to `CodeIgniter\Shield\Authentication\AuthenticatorInterface`:

```
/**
 * Starts login process
 */
public function startLogin(User $user): void;

/**
 * Completes login process
 */
public function completeLogin(User $user): void;
```

Useful for code auto-completion.

Use-case: dynamically log in a user when not going through the usual login process (login page, credentials verification, 2FA...).

That's it. Thanks

PS: I couldn't find a way to easily extend `AuthenticatorInterface` so I'm posting it here.

**Comment by grimpirate** [comment]
If you needed methods like that you could extend the Session Authenticator which includes those methods and then override them. It's what I did for my halberd extension for shield except that I didn't need to override those particular methods. More importantly, those methods may not be necessary for every authenticator so they shouldn't be included in the interface.
